src/zope/testrunner/options.py

```diff
@@ -27,17 +27,24 @@
 from zope.testrunner.formatter import terminal_has_colors
 from zope.testrunner.profiling import available_profilers
 
+def _regex_search(s):
+    return re.compile(s).search
 
-parser = optparse.OptionParser("Usage: %prog [options] [MODULE] [TEST]")
+parser = argparse.ArgumentParser(
+    description="Discover and run unittest tests")
+
+
+parser.add_argument("filters", nargs="*",
+                    help="Module or test filter")
```

Comment:
This should be

    parser.add_argument('module', nargs='?', dest='legacy_module_filter', help='Module filter')
    parser.add_argument('test', nargs='?', dest='legacy_test_filer', help='Test filter')

since there can be at most two, and the order is significant.

This will also allow us to simplify the handling of positional arguments somewhat (e.g. remove the branch that checks for more than two positional args being specified and prints an error).

Reply:
Great idea.